lib/auth/auth_with_roles.go

```diff
@@ -1484,7 +1485,279 @@ func (a *ServerWithRoles) GetNode(ctx context.Context, namespace, name string) (
 	return node, nil
 }
 
+func stringCompare(a string, b string, isDesc bool) bool {
+	if isDesc {
+		return a > b
+	}
+	return a < b
+}
+
+func unifiedNameCompare(a types.ResourceWithLabels, b types.ResourceWithLabels, isDesc bool) bool {
+	var nameA, nameB string
+	resourceA, ok := a.(types.Server)
+	if ok {
+		nameA = resourceA.GetHostname()
+	} else {
+		nameA = a.GetName()
+	}
+
+	resourceB, ok := b.(types.Server)
+	if ok {
+		nameB = resourceB.GetHostname()
+	} else {
+		nameB = b.GetName()
+	}
+
+	return stringCompare(nameA, nameB, isDesc)
+}
+
+func (s *ServerWithRoles) MakePaginatedResources(requestType string, resources []types.ResourceWithLabels) ([]*proto.PaginatedResource, error) {
+	paginatedResources := make([]*proto.PaginatedResource, 0, len(resources))
+	for _, resource := range resources {
+		var protoResource *proto.PaginatedResource
+		resourceKind := requestType
+		if requestType == types.KindUnifiedResource {
+			resourceKind = resource.GetKind()
+		}
+		switch resourceKind {
+		case types.KindDatabaseServer:
+			database, ok := resource.(*types.DatabaseServerV3)
+			if !ok {
+				return nil, trace.BadParameter("database server has invalid type %T", resource)
```

Comment:
These are prone to copy paste errors (congrats on not making one here!)

How about:

```suggestion
				return nil, trace.BadParameter("%v has invalid type %T", resourceKind, resource)
```

Then it's the same code in every case.